src/field.rs

```diff
@@ -772,6 +769,29 @@ pub fn random_vector<F: FieldElement>(len: usize) -> Result<Vec<F>, PrngError> {
     Ok(Prng::new()?.take(len).collect())
 }
 
+/// `encode_fieldvec` serializes a type that is equivalent to a vector of field elements.
+#[inline(always)]
+pub(crate) fn encode_fieldvec<F: FieldElement, T: AsRef<[F]>>(val: T, bytes: &mut Vec<u8>) {
+    for elem in val.as_ref() {
+        bytes.append(&mut (*elem).into());
+    }
+}
+
+/// `decode_fieldvec` deserializes some number of field elements from a cursor, and advances the
+/// cursor's position.
+pub(crate) fn decode_fieldvec<F: FieldElement>(
+    count: usize,
+    input: &mut Cursor<&[u8]>,
+) -> Result<Vec<F>, CodecError> {
+    let mut vec = Vec::with_capacity(count);
+    let mut buffer = vec![0; count * F::ENCODED_SIZE];
```

Comment:
Could we avoid an alloc by reading a fixed number of elements at a time?
```suggestion
    let mut buffer = [0; F::ENCODED_SIZE];
```

Reply:
It appears we can't yet, due to const limitations. `#![feature(generic_const_exprs)]` would partly unblock this.

```
error: constant expression depends on a generic parameter
   --> src/field.rs:787:26
    |
787 |     let mut buffer = [0; F::ENCODED_SIZE];
    |                          ^^^^^^^^^^^^^^^
    |
    = note: this may fail depending on what value the parameter takes
```

Reply:
Ooh right, I forgot :(

What about a fixed-length buffer, perhaps `[u8; 64]`? We're definitely never going to have a field that large.

Also, I might be over optimizing, so feel free to ignore.